internal/serialization/serialization.go

```diff
@@ -29,10 +29,20 @@ import (
 	"github.com/hazelcast/hazelcast-go-client/types"
 )
 
+// DefaultCompactDeserializer must be set before the client is created and should not be changed afterwards.
+var DefaultCompactDeserializer GenericCompactDeserializer
+
+// DefaultPortableDeserializer must be set before the client is created and should not be changed afterwards.
+var DefaultPortableDeserializer GenericPortableDeserializer
+
+// BuiltinDeserializers must be set before the client is created and should not be changed afterwards.
+var BuiltinDeserializers map[int32]pubserialization.Serializer
+
 // Service serializes user objects to Data and back to Object.
 // Data is the internal representation of binary Data in Hazelcast.
 type Service struct {
 	SerializationConfig  *pubserialization.Config
+	builtinSerializers   map[int32]pubserialization.Serializer
```

Comment:
Can't we use registry instead of creating this? Registry and builtinSerializers looks pretty similar in shape

Reply:
No, registry is for identified serializers, etc.

Reply:
In the future we may try to combine those maps, but it should work OK for now.